### PR TITLE
[TOSA] FFT2D/RFFT2D accuracy increased

### PR DESCRIPTION
**Pull request by d-smirnov**
This PR increases accurasy of FFT2D/RFFT2D calculation by removing periodic part of sin/cos

### CODE REVIEW

**Comment by d-smirnov** [comment]
@kuhar @MaheshRavishankar @Hsiangkai

File: mlir/lib/Conversion/TosaToLinalg/TosaToLinalg.cpp [hunk 1]
```diff
@@ -2364,16 +2365,24 @@ struct RFFT2dConverter final : public OpRewritePattern<RFFT2dOp> {
...
-      auto iyXoy = builder.create<arith::MulFOp>(loc, iy, oy);
-      auto ixXox = builder.create<arith::MulFOp>(loc, ix, ox);
-      auto yComponent = builder.create<arith::DivFOp>(loc, iyXoy, constH);
-      auto xComponent = builder.create<arith::DivFOp>(loc, ixXox, constW);
+      Value oy = builder.create<linalg::IndexOp>(loc, 1);
+      Value ox = builder.create<linalg::IndexOp>(loc, 2);
+      Value iy = builder.create<linalg::IndexOp>(loc, 3);
+      Value ix = builder.create<linalg::IndexOp>(loc, 4);
+
+      // float_t angle = 2 * pi() * ( ( (iy * oy) % H) / H + ( (ix * ox) % W ) /
```
Comment (GeorgeARM) [inline mlir/lib/Conversion/TosaToLinalg/TosaToLinalg.cpp:2373]: Can you please write a comment with what you are doing here?

Comment (d-smirnov) [reply mlir/lib/Conversion/TosaToLinalg/TosaToLinalg.cpp]: Amended

**Comment by d-smirnov** [comment]
@eric-k256

**Review by eric-k256 — APPROVED** [review]
Looks good to me.

**Pull request merged**